internal/api/v1beta1/domain.go

```diff
@@ -32,20 +33,27 @@ type DomainService interface {
 
 func (h Handler) CreateOrganizationDomain(ctx context.Context, request *frontierv1beta1.CreateOrganizationDomainRequest) (*frontierv1beta1.CreateOrganizationDomainResponse, error) {
 	logger := grpczap.Extract(ctx)
-
-	if request.GetOrgId() == "" || request.GetDomain() == "" {
-		return nil, grpcBadBodyError
+	orgResp, err := h.orgService.Get(ctx, request.GetOrgId())
+	if err != nil {
+		logger.Error(err.Error())
+		switch {
+		case errors.Is(err, organization.ErrNotExist):
+			return nil, grpcOrgNotFoundErr
+		default:
+			return nil, grpcInternalServerError
+		}
+	}
+	if orgResp.State == organization.Disabled {
+		return nil, grpcOrgDisabledErr
 	}
```

Comment:
What if we move this check to org service only and create another method `GetRaw` in org service. Org service will have two methods Get() and GetRaw().
1. Get() method will return the org only if its in enabled state, this is what we should use across all frontier
2. GetRaw will return the org no matter what the status is, this should only be used when we are explicitly hitting org API to get the org resource.

The reason I am proposing this change is avoid making mistake in future code changes where we forget to check if org is enabled or not and can lead to a security bypass. What do you think?